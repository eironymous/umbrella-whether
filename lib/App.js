"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _router = _interopRequireDefault(require("./router"));

var _reactRedux = require("react-redux");

var _routerSlice = require("./state/router-slice");

require("./App.css");

var _fontawesomeSvgCore = require("@fortawesome/fontawesome-svg-core");

var _freeSolidSvgIcons = require("@fortawesome/free-solid-svg-icons");

var _freeRegularSvgIcons = require("@fortawesome/free-regular-svg-icons");

var _state = _interopRequireDefault(require("./state/state"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

//Add fontawesome import to library
_fontawesomeSvgCore.library.add(_freeSolidSvgIcons.faUmbrella, _freeSolidSvgIcons.faSearch, _freeRegularSvgIcons.faEye, _freeSolidSvgIcons.faMinusCircle, _freeSolidSvgIcons.faHome, _freeSolidSvgIcons.faInfoCircle, _freeSolidSvgIcons.faQuestionCircle, _freeSolidSvgIcons.faHeart, _freeRegularSvgIcons.faHeart, _freeRegularSvgIcons.faSun, _freeSolidSvgIcons.faMoon, _freeSolidSvgIcons.faCloudSun, _freeSolidSvgIcons.faCloudMoon, _freeSolidSvgIcons.faCloudSunRain, _freeSolidSvgIcons.faCloudMoonRain, _freeSolidSvgIcons.faSmog, _freeSolidSvgIcons.faSnowflake, _freeSolidSvgIcons.faWind, _freeSolidSvgIcons.faCloudRain, _freeSolidSvgIcons.faBolt, _freeSolidSvgIcons.faCloudShowersHeavy, _freeSolidSvgIcons.faSurprise, _freeSolidSvgIcons.faStickyNote, _freeSolidSvgIcons.faEdit, _freeSolidSvgIcons.faEraser, _freeSolidSvgIcons.faPaperPlane, _freeSolidSvgIcons.faTrash);

app.use(express.static(path.join(__dirname, 'public')));
app.get('*', function (req, res) {
  res.sendFile(path.join(__dirname, 'public', 'index.html'));
});

var AppWrapper = function AppWrapper() {
  //Wrap app in a provider
  return <_reactRedux.Provider store={_state.default}>
      <App />
    </_reactRedux.Provider>;
};

function App() {
  var dispatch = (0, _reactRedux.useDispatch)();
  dispatch((0, _routerSlice.updateRoute)("home"));
  return <div className="App">
        <_router.default />
    </div>;
}

var _default = AppWrapper;
exports.default = _default;