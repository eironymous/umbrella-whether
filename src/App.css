.App {
  text-align: center;
  background-color: #22223B;
  color: rgb(255, 255, 255);
  min-height: 100vh;
  box-sizing: border-box;
  overflow: hidden;
  min-width: 1200px;
}

