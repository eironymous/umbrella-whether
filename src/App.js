import React from 'react';
import Router from "./router";
import './App.css';
import { library } from "@fortawesome/fontawesome-svg-core";
import { faUmbrella, faSearch, faMinusCircle, faHome, faInfo, faQuestion, faHeart as fullHeart } from "@fortawesome/free-solid-svg-icons";
import { faEye, faHeart as emptyHeart } from "@fortawesome/free-regular-svg-icons";
import { Provider } from "react-redux";
import store from "./state/state";

//Add fontawesome import to library
library.add(faUmbrella, faSearch, faEye, faMinusCircle, faHome, faInfo, faQuestion, fullHeart, emptyHeart);

const AppWrapper = () => {
  //Wrap app in a provider
  return (
    <Provider store={store}>
      <App/>
    </Provider>
  )
}

function App() {
  return (
    <div className="App">
        <Router />
    </div>
  );
}

export default AppWrapper;