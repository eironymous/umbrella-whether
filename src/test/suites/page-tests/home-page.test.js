import React from "react";
import App from "../../../App";
import Home from "../../../pages/home/index";
import { mount, shallow, configure } from "enzyme";
import { Provider } from "react-redux";
import Adapter from "enzyme-adapter-react-16";
import store from "../../../state/state";

configure({ adapter: new Adapter() });

// test('App loads correctly', () => {
// 	let wrapper = mount(<App/>);

// 	expect(wrapper).toMatchSnapshot();
// })

// test('Home page loads correctly', () => {
// 	let wrapper = mount(
// 		<Provider store={store}>
// 			<Home activeRoute={"home"} allRoutes={"home"} />
// 		</Provider>
// 	);
	
// 	expect(wrapper.find("h1").text()).toContain("home")
// });