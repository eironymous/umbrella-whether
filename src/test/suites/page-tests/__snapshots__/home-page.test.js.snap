// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App loads correctly 1`] = `ReactWrapper {}`;
